rasa/core/training/story_conflict.py

```diff
@@ -0,0 +1,292 @@
+        Creates a `StoryConflict` from a given state.
+
+        Args:
+            sliced_states: The (sliced) dialogue state at which the conflict occurs.
+        """
+        self._sliced_states = sliced_states
+        self._conflicting_actions = defaultdict(
+            list
+        )  # {"action": ["story_1", ...], ...}
+
+    def __hash__(self) -> int:
+        return hash(str(list(self._sliced_states)))
+
+    def add_conflicting_action(self, action: Text, story_name: Text) -> None:
+        """Adds another action that follows from the same state.
+
+        Args:
+            action: Name of the action.
+            story_name: Name of the story where this action is chosen.
+        """
+        self._conflicting_actions[action] += [story_name]
+
+    @property
+    def conflicting_actions(self) -> List[Text]:
+        """List of conflicting actions.
+
+        Returns:
+            List of conflicting actions.
+
+        """
+        return list(self._conflicting_actions.keys())
+
+    @property
+    def conflict_has_prior_events(self) -> bool:
+        """Checks if prior events exist.
+
+        Returns:
+            `True` if anything has happened before this conflict, otherwise `False`.
+        """
+        return _get_previous_event(self._sliced_states[-1])[0] is not None
+
+    def __str__(self) -> Text:
+        # Describe where the conflict occurs in the stories
+        last_event_type, last_event_name = _get_previous_event(self._sliced_states[-1])
+        if last_event_type:
+            conflict_message = (
+                f"CONFLICT after {last_event_type} '{last_event_name}':\n"
+            )
+        else:
+            conflict_message = f"CONFLICT at the beginning of stories:\n"
+
+        # List which stories are in conflict with one another
+        for action, stories in self._conflicting_actions.items():
+            conflict_message += f"  {self._summarize_action_occurence(action, stories)}"
+
+        return conflict_message
+
+    @staticmethod
+    def _summarize_action_occurence(action: Text, stories: List[Text]) -> Text:
+        """Gives a summarized textual description of where one action occurs.
+
+        Args:
+            action: The name of the action.
+            stories: The stories in which the action occurs.
+
+        Returns:
+            A textural summary.
+        """
+        if len(stories) > 3:
+            # Four or more stories are present
+            conflict_description = (
+                f"'{stories[0]}', '{stories[1]}', and {len(stories) - 2} other trackers"
+            )
+        elif len(stories) == 3:
+            conflict_description = f"'{stories[0]}', '{stories[1]}', and '{stories[2]}'"
+        elif len(stories) == 2:
+            conflict_description = f"'{stories[0]}' and '{stories[1]}'"
+        elif len(stories) == 1:
+            conflict_description = f"'{stories[0]}'"
+        else:
+            raise ValueError("Trying to summarize conflict without stories.")
```

Comment:
can we give the user a hint how they can fix this?

Reply:
This should actually be impossible to happen. If it happens, then it is a bug. So I'd write something like "An internal error occurred while trying to summarise a conflict without stories. Please file a bug report at github.com/..."